### PR TITLE
added conditional filter (skips if contains words) Sent Tweet to DM

### LINKED ISSUE #5: Retweeted an Advert
Miss on retweeting advert
https://twitter.com/CupidName/status/1245185624031801350

### PR DESCRIPTION
**Pull request by skyhoshi**
I added an additional filter list (words that if found should mark the tweet as not retweetable). 
just in case it should send the tweets to the DM's for review.

Summary of the changes

- Added string "Filter"
- tested content of the tweet for the filter words.

Fixes #5

### CODE REVIEW

**Comment by skyhoshi** [comment]
I was NOT able to test this as I don't have an AWS account.

File: src/DotNetTwitterBot/Function.cs [hunk 1]
```diff
@@ -20,10 +21,12 @@ public async Task Retweet(ILambdaContext context)
 
             Auth.SetUserCredentials(creds.ConsumerKey, creds.ConsumerSecret, creds.AccessToken, creds.AccessSecret);
 
-            var searchTerms = new[] { "\".NET Framework\"", "\".NET Core\"", "\".NET 5\"", "dotnet", "dotnetcore", "_dotnetbot_" };
+            var searchTerms = new[] { $"\".NET Framework\"", "\".NET Core\"", "\".NET 5\"", "dotnet", "dotnetcore", "_dotnetbot_" };
```
Comment (marcusturewicz) [inline src/DotNetTwitterBot/Function.cs:24]: The `$` is not necessary here, there's no string interpolation being done.

File: src/DotNetTwitterBot/Function.cs [hunk 2]
```diff
@@ -20,10 +21,12 @@ public async Task Retweet(ILambdaContext context)
...
 
             var searchSince = DateTime.UtcNow.Subtract(TimeSpan.FromMinutes(10));
 
+            var filterTerms = new[] { $@"domain", $@"registration", $@"domainregistration" };
```
Comment (marcusturewicz) [inline src/DotNetTwitterBot/Function.cs:28]: The `$@` are not necessary here.

**Comment by skyhoshi** [comment]
you are totally correct on those nits. :) I was trying things out and forgot to remove the one from the top and the rest are just form of habit. 

I'll edit those ASAP (tonight, expect it soon)

File: src/DotNetTwitterBot/Function.cs [hunk 3]
```diff
@@ -43,6 +46,13 @@ public async Task Retweet(ILambdaContext context)
                     if (!tweet.Text.Contains(term, StringComparison.OrdinalIgnoreCase))
                         continue;
 
+                    // Exclude tweets that contain excluded words.
+                    if (filterTerms.Any(d => tweet.Text.Contains(d)))
+                    {
+                        //This sends the questionable tweet to the DM's of the me.User
+                        Message.PublishMessage($"Questionable Tweet : {tweet.Url}", User.GetUserFromScreenName($"{me.ScreenName}").Id);
```
Comment (marcusturewicz) [inline src/DotNetTwitterBot/Function.cs:53]: Sorry, missed this before. Is there an async version of `PublishMessage`? I would prefer to use that:
```csharp
await Message.PublishMessageAsync(...);
```

Comment (skyhoshi) [reply src/DotNetTwitterBot/Function.cs:53]: no problem

File: src/DotNetTwitterBot/Function.cs (hunk 1 above)
Comment (skyhoshi) [reply src/DotNetTwitterBot/Function.cs]: done

File: src/DotNetTwitterBot/Function.cs (hunk 2 above)
Comment (skyhoshi) [reply src/DotNetTwitterBot/Function.cs]: done

File: src/DotNetTwitterBot/Function.cs (hunk 3 above)
Comment (skyhoshi) [reply src/DotNetTwitterBot/Function.cs]: done

File: src/DotNetTwitterBot/Function.cs [hunk 4]
```diff
@@ -43,6 +46,13 @@ public async Task Retweet(ILambdaContext context)
                     if (!tweet.Text.Contains(term, StringComparison.OrdinalIgnoreCase))
                         continue;
 
+                    // Exclude tweets that contain excluded words.
+                    if (filterTerms.Any(d => tweet.Text.Contains(d)))
+                    {
+                        //This sends the questionable tweet to the DM's of the me.User
+                        Message.PublishMessageAsync($"Questionable Tweet : {tweet.Url}", User.GetUserFromScreenName($"{me.ScreenName}").Id);
```
Comment (marcusturewicz) [inline src/DotNetTwitterBot/Function.cs:53]:
```suggestion
                        await Message.PublishMessageAsync($"Questionable Tweet : {tweet.Url}", User.GetUserFromScreenName($"{me.ScreenName}").Id);
```

Comment (skyhoshi) [reply src/DotNetTwitterBot/Function.cs:53]: oh man, :D

**Pull request merged**